extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -285,6 +303,49 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
         this.win, DOUBLECLICK_EXPERIMENT_FEATURE.DELAYED_REQUEST);
   }
 
+  /** @override */
+  buildCallback() {
+    super.buildCallback();
+    if (this.win['dbclk_a4a_viz_change']) {
```

Comment:
Will address in a subsequent PR as this is a larger change.  I agree that having a dedicate service for state is a good idea.